velox/substrait/SubstraitToVeloxExpr.cpp

```diff
@@ -29,17 +29,39 @@ SubstraitVeloxExprConverter::toVeloxExpr(
         kDirectReference: {
       const auto& directRef = substraitField.direct_reference();
       int32_t colIdx = substraitParser_.parseReferenceSegment(directRef);
+
+      const auto& inputTypes = inputType->children();
       const auto& inputNames = inputType->names();
       const int64_t inputSize = inputNames.size();
-      if (colIdx <= inputSize) {
-        const auto& inputTypes = inputType->children();
-        // Convert type to row.
+      VELOX_CHECK_LT(
+          colIdx, inputSize, "Missing the column with id '{}' .", colIdx);
+      std::optional<int32_t> childIdx;
+      if (directRef.struct_field().has_child()) {
+        childIdx = substraitParser_.parseReferenceSegment(
+            directRef.struct_field().child());
+      }
+
+      if (!childIdx.has_value()) {
         return std::make_shared<core::FieldAccessTypedExpr>(
             inputTypes[colIdx],
             std::make_shared<core::InputTypedExpr>(inputTypes[colIdx]),
             inputNames[colIdx]);
+      }
+
+      // Select a subfield in a struct by name.
+      if (auto inputColumnType = asRowType(inputTypes[colIdx])) {
+        VELOX_CHECK_LT(
+            childIdx.value(),
+            inputColumnType->size(),
+            "Missing the subfield with id '{}' .",
+            childIdx.value());
+        return std::make_shared<core::FieldAccessTypedExpr>(
+            inputColumnType->childAt(childIdx.value()),
+            std::make_shared<core::FieldAccessTypedExpr>(
+                inputTypes[colIdx], inputNames[colIdx]),
+            inputColumnType->nameOf(childIdx.value()));
       } else {
-        VELOX_FAIL("Missing the column with id '{}' .", colIdx);
+        VELOX_FAIL("RowType expected.");
```

Comment:
It would be easier to read if this check was moved earlier and "if" is removed.

```
VELOX_CHECK_EQ(inputTypes[colIdx]->typeKind(), TypeKind::ROW);
```